tests/test_safety.py

```diff
@@ -204,7 +204,34 @@ def test_check_live_cached(self):
         )
         self.assertEqual(len(vulns), 1)
 
+    def test_check_live_down_cached(self):
+        reqs = StringIO("insecure-package==0.1")
+        packages = util.read_requirements(reqs)
 
+        vulns = safety.check(
+            packages=packages,
+            db_mirror=False,
+            cached=True,
+            key=False,
+            ignore_ids=[],
+            proxy={}
+        )
+        self.assertEqual(len(vulns), 1)
+        
+        reqs = StringIO("insecure-package==0.1")
+        packages = util.read_requirements(reqs)
+        # make a second call, where server responds with 400 error, so use
+        # the cache
+        vulns = safety.check(
+            packages=packages,
+            db_mirror="http://pyup.io/",
```

Comment:
Can you use **mock** to avoid letting **requests** performing an actual HTTP request?